common/test/acceptance/tests/studio/test_studio_outline.py

```diff
@@ -1578,3 +1583,139 @@ def _get_items(self):
+                                   'from the course advanced settings. To do this, go to the Advanced Settings '
+                                   'page, locate the "Advanced Module List" setting, and then delete the following '
+                                   'modules from the list.')
+
+    def _add_deprecated_advance_modules(self, block_types):
+        """
+        Add `block_types` into `Advanced Module List`
+
+        Arguments:
+            block_types (list): list of block types
+        """
+        self.advanced_settings.visit()
+        self.advanced_settings.set_values({"Advanced Module List": json.dumps(block_types)})
+
+    def _create_deprecated_components(self):
+        """
+        Create deprecated components.
+        """
+        parent_vertical = self.course_fixture.get_nested_xblocks(category="vertical")[0]
+
+        self.course_fixture.create_xblock(
+            parent_vertical.locator,
+            XBlockFixtureDesc('combinedopenended', "Open", data=load_data_str('ora_peer_problem.xml'))
+        )
+        self.course_fixture.create_xblock(parent_vertical.locator, XBlockFixtureDesc('peergrading', 'Peer'))
+
+    def _verify_deprecation_warning_info(
+            self,
+            deprecated_blocks_present,
+            components_present,
+            components_display_name_list=None,
+            deprecated_modules_list=None
+    ):
+        """
+        Verify deprecation warning
+
+        Arguments:
+            deprecated_blocks_present (bool): advance modules remove text is visible if True else False
+            components_present (bool): components list is visible if True else False
+            components_display_name_list (list): list of components display name
+            deprecated_modules_list (list): list of deprecated advance modules
+        """
+        self.assertTrue(self.course_outline_page.deprecated_warning_visible)
+        self.assertEqual(self.course_outline_page.warning_heading_text, self.HEADING_TEXT)
+        self.assertEqual(self.course_outline_page.modules_remove_text_shown, deprecated_blocks_present)
+        if deprecated_blocks_present:
+            self.assertEqual(self.course_outline_page.modules_remove_text, self.ADVANCE_MODULES_REMOVE_TEXT)
+            self.assertEqual(self.course_outline_page.deprecated_advance_modules, deprecated_modules_list)
+
+        self.assertEqual(self.course_outline_page.components_visible, components_present)
+        if components_present:
+            self.assertEqual(self.course_outline_page.components_list_heading, self.COMPONENT_LIST_HEADING)
+            self.assertItemsEqual(self.course_outline_page.components_display_names, components_display_name_list)
+
+    def test_no_deprecation_warning_message_present(self):
+        """
+        Scenario: Verify that deprecation warning message is not shown if ORA1 advance modules are not present and
```

Comment:
Nit: docstrings are wrapped at 80 characters, and wrapped lines are only indented four spaces.